systemd/systemd.go

```diff
@@ -563,6 +614,51 @@ func (s *systemd) StartNoBlock(serviceNames []string) error {
 	return err
 }
 
+// Check if unit is in failed state
+func (s *systemd) IsFailed(serviceNames []string) bool {
+	var err error
+	var out []byte
+	if s.rootDir != "" {
+		out, err = s.systemctl(append([]string{"--root", s.rootDir, "is-failed"}, serviceNames...)...)
+	} else {
+		out, err = s.systemctl(append([]string{"is-failed"}, serviceNames...)...)
+	}
+	if err != nil {
+		// is-failed returns non-0 if the unit is not in the failed
+		// state, eg. active, inactive etc.
+		return false
+	}
+
+	// failed is reported only for units in the failed state
+	return "failed" == strings.TrimSpace(string(out))
```

Comment:
nitpick: not sure how our (or golang's) coding style goes here, but I find it more readable when the first operand at the left of `==` is the mutable value.